### ISSUE TITLE
[no-use-before-define] False positive on type guard cast

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#2527]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-use-before-define": "error"
  }
}
```

```TS
import { isNumber } from 'type-guards';
type Id = number;

const isId = isNumber as (value: unknown) => value is Id;
```

**Expected Result**

No error, `value` is part of the type, it's not being "used".

**Actual Result**

```ts
const isId = isNumber as (value: unknown) => value is Id;
//                                           ~~~~~ 'value' was used before it was defined. (eslint@typescript-eslint/no-use-before-define)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `12.18.1` |

**Comment by bradzacher** [comment]
This is a fun bug because technically `(value: unknown)` creates a "value" variable in the function type's scope (`x is y` and `typeof x` both require that `x` is a "value" variable, not a "type" variable).

So the rule is seeing a "value" variable with no initialisation and reporting. Should be a pretty easy fix.
